nodejs/week3/week3/warmup/index.js

```diff
@@ -0,0 +1,99 @@
+
+
+app.get("/calculator/multiply", (req, res) => {
+   
+    const firstNumber= Number(req.query.firstParam);
+    const secondParam=req.query.secondParam;
+   
+    
+    if(typeof(secondParam) === 'object'){
+       
+        const secondParamArray=secondParam.reduce( function (previous,current) {
+         return Number(previous)*Number(current)});
+        
+         const result1=firstNumber * secondParamArray;
+        res.json(`The answer for your request is ${result1}`);
+    }else{
+        const secondNumber=Number(req.query.secondParam);
+        const result =firstNumber * secondNumber;
+        res.json(`The answer for your request is ${result}`);
+    }   
+   
+});
+
+app.get("/calculator/add", (req, res) => {
+   
+    const firstNumber= Number(req.query.firstParam);
+    const secondNumber=Number(req.query.secondParam);
+    const thirdNumber =Number(req.query.thirdParam);
+    if(thirdNumber){
+        const result1=firstNumber + secondNumber + thirdNumber;
+        res.json(`The answer for your request is ${result1}`);
+    }else{
+        const result =firstNumber + secondNumber;
+    res.json(`The answer for your request is ${result}`);
+    }
+
+});
+
+app.get("/calculator/division", (req, res) => {
+    
+    const firstNumber= Number(req.query.firstParam);
+    const secondNumber=Number(req.query.secondParam);
+    const result =firstNumber / secondNumber;
+    res.json(`The answer for your request is ${result}`);
+    
+});
+
+app.get("/calculator/Subtraction", (req, res) => {
+    
+    const firstNumber= Number(req.query.firstParam);
+    const secondNumber=Number(req.query.secondParam);
+    const result =firstNumber - secondNumber;
+    res.json(`The answer for your request is ${result}`);
+    
+});
+
+app.post("/calculator/multiply", (req, res) => {
+      
+    const firstNumber= Number(req.body.firstParam);
+    const secondNumber=Number(req.body.secondParam);
+    const result = firstNumber * secondNumber;
+    res.json(`The answer for your request is ${result}`);
+   
+});
+
+app.post("/calculator/add", (req, res) => {
+    
+    const firstNumber= Number(req.body.firstParam);
+    const secondNumber=Number(req.body.secondParam);
+    const result =firstNumber + secondNumber;
+    res.json(`The answer for your request is ${result}`);
+   
+});
+
+app.post("/calculator/division", (req, res) => {
+    
+    const firstNumber= Number(req.body.firstParam);
+    const secondNumber=Number(req.body.secondParam);
+    const result =firstNumber / secondNumber;
+    res.json(`The answer for your request is ${result}`);
+  
+});
+
+app.post("/calculator/Subtraction", (req, res) => {
+    
+    const firstNumber= Number(req.body.firstParam);
+    const secondNumber=Number(req.body.secondParam);
+    const result =firstNumber - secondNumber;
+    res.json(`The answer for your request is ${result}`);
+   
+});
+
+app.listen(3000, () => console.log(`Calculator:listening on port 3000`));
```

Comment:
@Vahablotfi Use the array method reduce to put all the second params together just as you did with the multiply in the first example, because in as this will work because you're testing with only two numbers/params, if I add a third,fourth,etc it'll fail.